pkg/metrics/metrics.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+type MetricDefinition struct {
+	Name   string
+	Help   string
+	Type   metricType
+	Labels []string
+}
+
+type PromConnectionInfo struct {
+	Address string
+	Port    int
+}
+
+type Settings struct {
+	PromConnectionInfo
+	Prefix string
+}
+
+type metricType string
+
+const (
+	TypeCounter   metricType = "counter"
+	TypeGauge     metricType = "gauge"
+	TypeHistogram metricType = "histogram"
+)
+
+var allMetrics = []MetricDefinition{}
+
+func defineMetric(name, help string, t metricType, labels ...string) MetricDefinition {
+	def := MetricDefinition{
+		Name:   name,
+		Help:   help,
+		Type:   t,
+		Labels: labels,
+	}
+	allMetrics = append(allMetrics, def)
+	return def
+}
+
+var (
+	hmapEvictionsTotal = defineMetric(
```

Comment:
I have the feeling that we could merge metrics used for ringbuf with ones used for maps.
E.g. we could have:
- evictions_total{source=bpf_ringbuf | bpf_maps}
- evicted_flows_total{source=bpf_ringbuf | bpf_maps}

That would replace hmapEvictionsTotal, userspaceNumberOfEvictionsTotal, numberOfevictedFlowsTotal and numberofFlowsreceivedviaRingBufferTotal

Reply:
there is either timer eviction for hmap or event for rb not different reasons for evictions from looking at the code